File: src/code/Insertion.java
package code;
import java.util.ArrayList;
import java.util.Arrays;

/**
 * Generic Class Insertion Initialization With Implemented AlgorithmBasics
 */
public class Insertion<Type extends Comparable<Type>> implements AlgorithmBasics<Type>
{
    // ArrayList Variable
    private ArrayList<Type> unsortedArray;
    
    // Default Constructor 
    public Insertion()
    {
        unsortedArray = new ArrayList<Type>();
    }

    // Constructor With Already Made Array As Parameter
    public Insertion(ArrayList<Type> unsortedArray)
    {
        this.unsortedArray = unsortedArray;
    }

    /**
     * Sorting Method Which Sorts The Array Using Insertion Sort
     * @return returns the object
     */
    public Insertion<Type> sort()
    {
        
        Type key;
        int j;
        for(int i=1; i<unsortedArray.size(); i++)
        {
            // The Furthest Index To The Right Of The Already Sorted Portion
            j = i-1;
            // The Value We Want To Add To The Already Sorted Portion
            key = unsortedArray.get(i);

            // This While Loop Keeps Checking The Already Sorted Portion To Determine Where Key Should Be Placed
            while(j>=0 && key.compareTo(unsortedArray.get(j)) == -1)
            {
                // Keeps Shifting Any Value Greater Than Key To The Right
                unsortedArray.set(j+1, unsortedArray.get(j));
                // Allows Us To Keep Comparing To The Left
                j-=1;
            }
            // Finally We Place The Key In Its Correct Spot
            unsortedArray.set(j+1, key);
        }
        return this;
    }

    /**
     * Gives Access To Private ArrayList
     * @return returns the arraylist
     */
    public ArrayList<Type> getUnsortedArray()
    {
        return unsortedArray;
    }

    /**
     * Add Another Value To The ArrayList
     * @param value the value to be added
     */
    public void addUnsortedArray(Type value)
    {
        unsortedArray.add(value);
    }

    public ArrayList<Type> getArray()
    {
        return unsortedArray;
    }

    /**
     * Displays The Array Instead Of Memory Address
     * @return returns the array as a string
     */
    public String toString()
    {
        return String.valueOf(unsortedArray);
    }

    // Main Method
    public static void main(String[] args)
    {
        Insertion<Integer> newInsertion = new Insertion<Integer>(new ArrayList<Integer>(Arrays.asList(12, 2, 5, 1, 6, 2, 10)));
        System.out.println("Unsorted: " + newInsertion.getUnsortedArray());
        newInsertion.sort();
        System.out.println("Sorted: " + newInsertion.getUnsortedArray());
    }
    
}

File: src/code/Insertion.py
import random

# Class Insertion
class Insertion:
    
    # Default Constructor
    # Sets The List With A Length Of 5 Random Numbers
    def __init__(self):
        self.unsortedList = [random.randint(0, 21) for i in range(5)]
    # Method Sort
    # Sorts The List Using Insertion Sort Algorithm
    def sort(self):
        # Begin Iterating From Index 1 To The Remaining Length Of The List
        for i in range(1, len(self.unsortedList)):
            # Key Is The Data We Want To Have Inserted Into The Already Sorted Portion Of The List
            key = self.unsortedList[i]
            # J Is The Index We Will Start Comparing With To See If Swapping Is Needed
            j = i-1

            # Checking To See If It Doesn't Cause Index Out Of Bounds With -1
            # Checking If Key Value Is Less Than The Already Sorted Values
            while(j>=0 and key < self.unsortedList[j]):
                # Sets The Index Ahead To The Same Value
                self.unsortedList[j+1] = self.unsortedList[j]
                # Brings J By 1 Index To Compare The Next Lower Index
                j -= 1
            # Once Loop Is Done Key Was Holding The Temporary Value Now Equals That Index
            self.unsortedList[j+1] = key
    # Displays Object As String Rather Than Object Memory Address
    def __str__(self):
        return f"{self.unsortedList}"

if __name__ == "__main__":
    unsortedList = Insertion()
    print(f"Unsorted List: {unsortedList}")
    unsortedList.sort()
    print(f"Sorted List: {unsortedList}")